modules/math/shared/src/main/scala/gsp/math/skycalc/solver/Interval.scala

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) 2016-2020 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gsp.math.skycalc.solver
+
+import cats.implicits._
+import cats.Eq
+import cats.Order
+import cats.Show
+import java.time.Duration
+import java.time.Instant
+import java.time.LocalTime
+import java.time.ZoneId
+import monocle.Getter
+import monocle.Prism
+import gsp.math.optics.Format
+import io.chrisdavenport.cats.time._
+
+/**
+  * Representation of an interval between two points in time, including the start time and excluding the end time
+  * (i.e. the interval [start, end)).
+  *
+  * Note that an interval can not represent a single point in time (i.e. start == end) because such an interval
+  * could not contain any time t for which t >= start && t < end.
+  */
+sealed abstract case class Interval protected (start: Instant, end: Instant) {
```

Comment:
It would be nice if we could replace this with `spire.math.Interval` in a followup PR.

Reply:
I'm taking a look at Spire. We could definitely use that. It'd be nice if the bound types (open or closed) were encoded in the type, since here we want to enforce [start, end).